### ISSUE TITLE
Model Zoo ".caffemodel" use with digits

### ISSUE THREAD

**Issue by Pessanha24** [NVIDIA/DIGITS#720]
I download the "train_val.prototxt" and the "*.caffemodel" file from the Model Zoo site.

I copy and paste the "train_val.prototxt" content to the "custom Network" area adding a "#" to the "source" and "backend" lines.

I specify the path to the folder were I have the "*.caffemodel" file but I obtain the error:
"Pretrained model
File does not exist"

What I am doing wrong? I didn't found proper support for this use, can you help me?

Thanks in advance.

**Comment by lukeyeager** [comment]
> I specify the path to the folder were I have the "*.caffemodel" file

I assume you mean you specified the path to the **file**, and not just the file's parent directory?

**Comment by Pessanha24** [comment]
Yes, I specify the path to the file and I'm not understanding what I am doing wrong.

Can you confirm me that I only need to add "#" to the "source" and "backend" lines of  "train_val.prototxt" and copy it to the Custom Network field and them specify the path to the caffemodel file?

![error](https://cloud.githubusercontent.com/assets/8130695/14998897/f9d2806e-117d-11e6-9e69-b6e76b4db8b6.png)

**Comment by lukeyeager** [comment]
That path looks suspicious to me - you're sure you didn't mean to include your username?

    /home/<username>/squeezenet_v1.0.caffemodel

**Comment by Pessanha24** [comment]
I solve the problem downloading again the file from the [caffe model zoo ](https://github.com/BVLC/caffe/wiki/Model-Zoo) and choosing another directory. I am not sure of what was the solution, but now it works.

Now I have the "ERROR: must specify a loss layer" with a network trained for imagenet.  Do you now why?

**Comment by lukeyeager** [comment]
Click on the little blue question mark next to "Custom Network". That should tell you everything you need to know - let me know if it doesn't and we'll update the documentation!

**Comment by Pessanha24** [comment]
I'm trying to use [this](https://gist.github.com/jimmie33/339fd0a938ed026692267a60b44c0c58) pre-trained model  with the ".caffemodel" and "deploy.prototxt" file available to initialize my network weights to be able to find a better classifier for my dataset.

I didn't understand how I add the "SoftmaxWithLoss" layer for this example. Can you help me?

**Comment by lukeyeager** [comment]
The model description claims that it's an object detection network, but it only has a `1x100` shape output? Not sure what's going on there. Are you sure you want to repurpose this network for image classification?

If so, you probably want to remove the `Sigmoid` layer and replace it with 3 layers that activate on different phases, as with all the standard networks:
https://github.com/NVIDIA/DIGITS/blob/v3.3.0/digits/standard-networks/caffe/lenet.prototxt#L156-L178
https://github.com/NVIDIA/DIGITS/blob/v3.3.0/digits/standard-networks/caffe/alexnet.prototxt#L365-L387
https://github.com/NVIDIA/DIGITS/blob/v3.3.0/digits/standard-networks/caffe/googlenet.prototxt#L2474-L2508

**Comment by Pessanha24** [comment]
"Are you sure you want to repurpose this network for image classification?"

I just want to learn how to use another previously trained networks (pre-trained weights) to use them with my dataset and increase the quality of my classifier. That was the reason I went to the caffe Model Zoo website because I think that the application was more direct in Digits. But I verified that most of them didn't work directly without some previous adaptations.

I didn't test your recommendation from your last answer yet ("3 layers ") but I will make it as soon as I can.

**Comment by Pessanha24** [comment]
For example with the [SqueezeNet](https://github.com/DeepScale/SqueezeNet) it works in a simple way to, just adding it to the DIGITS.

![squeeze](https://cloud.githubusercontent.com/assets/8130695/15099833/bef67854-1558-11e6-8dfd-acba69f266e7.png)
 
The main difference here is the three layers described in the Figure that are not present in some of the models available in the model Zoo website.

**Comment by lukeyeager** [comment]
Those layers are typical for classification networks. For solving other problems, you'll probably want to use the "Generic"/"Other" path - that will work for networks which don't have a `Softmax` layer output.
